src/share/classes/com/sun/javatest/regtest/report/RegressionReporter.java

```diff
@@ -111,12 +111,12 @@ public void report(RegressionParameters params, ElapsedTimeHandler elapsedTimeHa
                 if (testStats != null)
                     testStats.report(r);
 
-                for (var sr : List.of(
-                        SummaryReporter.forTestNG(params.getWorkDirectory()),
-                        SummaryReporter.forJUnit(params.getWorkDirectory()))) {
-                    if (!sr.isEmpty()) {
-                        sr.writeReport(rd);
-                    }
+                int countTestNG = SummaryReporter.forTestNG(params.getWorkDirectory()).writeReport(rd);
+                int countJUnit = SummaryReporter.forJUnit(params.getWorkDirectory()).writeReport(rd);
+                int sumOfCounts = countTestNG + countJUnit;
+                if (sumOfCounts > 0) {
+                    log.println(String.format("Framework-based tests: %d, TestNG: %d, JUnit: %d",
+                            sumOfCounts, countTestNG, countJUnit));
```

Comment:
I'd skip printing `sumOfCounts` and only print the individual testNG and JUnit counts; without looking at the code it's not clear to me that "Framework-based tests" is a sum of the two.

While at it, any chance to print the number of skipped test cases in JUnit / testNG?

Reply:
> `Framework-based tests: 188 = 104 TestNG + 84 JUnit`

That's much better IMO. Less ambiguous.

Thanks for the pointer to the text files, I'll check them out.